src/util/color.hpp

```diff
@@ -35,24 +33,18 @@ constexpr auto bit_shift = sizeof(ColorInt) * 2;
 // 1 for the size of the color channel
 constexpr ColorInt bits_place = (1 << bit_shift) - 1;
 
-/*inline ColorFloat
+inline ColorFloat
 convert_color_data(ColorInt int_color) {
-    uint32_t red = (int_color >> bit_shift * 3) & bits_place;
-    uint32_t green = (int_color >> bit_shift * 2) & bits_place;
-    uint32_t blue = (int_color >> bit_shift) & bits_place;
-    uint32_t alpha = int_color & bits_place;
-    // the last one >> 0 is A
-    ColorFloat vector_color(
-        {red / 255.0f, green / 255.0f, blue / 255.0f, alpha / 255.0f}
-    );
-    return vector_color;
-}*/
+    glm::vec4 color = ((glm::vec4)uint32_to_color(int_color)) / 255.0f;
+    return {color.a, color.b, color.g, color.r};
```

Comment:
```suggestion
    auto color = uint32_to_color(int_color);
    return color / 255.0f;
```

Reply:
Did you just ignore this comment?

Reply:
I mean it's not right. `vec4` is stored as rgba, so the way that you converted switches the order.

Reply:
A json file, and a qb file. The data is saved as 8 for R, 8 for G, 8 for B, 8 for A.

Reply:
Go swap these conditions to `std::endian::big`:
- https://github.com/AlemSnyder/Fun-Game/blob/544303ae54fd6e1cdd67559e070cabeebec28eb4/src/util/voxel_io.hpp#L44-L47
- https://github.com/AlemSnyder/Fun-Game/blob/544303ae54fd6e1cdd67559e070cabeebec28eb4/src/util/voxel_io.hpp#L35-L38

Then fix this.
I'm amazed you don't see the issue. That the int colors are using a different representation than the floats. So instead of fixing the root issue (we parse the file wrong). Your solution is really to make this bandaid over it.

Reply:
It worked fine, then you changed it and it broke. Then I fixed it, and now you say it is wrong. How about you make a pr I will compile and test. You may "fix" it again, but not in this pr.

Reply:
Ok I pushed a commit that should fix it.